release/models/system/openconfig-system-controlplane.yang

```diff
@@ -198,6 +203,31 @@ module openconfig-system-controlplane {
                 }
               }
             }
+
+            container queue {
+              description
+                "Configuration and operational state relating to the QoS
+                queues that are used for control-plane traffic.  The
+                control-plane container is treated like an interface.";
+
+              container config {
```

Comment:
I'm not sure if this is necessary to configure queue name. But it doesn't not harm either.

Reply:
This leaf is added so one can stream a list of queues, keyed by name.  I expect systems will only expose these as state and not allow configuration as I understand these names are typically fixed based on the hardware that is used.

Reply:
I see your point though, if we expect the control traffic queue names are statically defined by the system, we could make the name state only.   My guess is like you said, leave name as configurable as it is not harmful, even if the referenced platforms do not support it currently.  @robshakir any thoughts?

Reply:
In the QoS model we have this a read-write leaf because there are other parameters that are being specified under this queue (per-interface per-queue queue-management approach for example). If we don't foresee a need here that this will ever be configurable in this context then a ro leaf is possible.

I don't think this is the same as the discussion above though - the classifier that is applied here can specify a group to queue mapping, which may be more than just the set of queues that are available in hardware. For example, CoPP policies can have concepts such as the "configured BGP peers" in a queue with a specific rate limit etc. This however, would be done through the classifier and the queue counters would still be here with that static name.

Reply:
For a moment I would like to focus on what is the intended way to count policer drops for control plane traffic, since that is the operational goal.   Is my interpretation below correct?

`/qos/scheduler-policies/scheduler-policy/schedulers/scheduler/one-rate-two-color/config/queuing-behavior` is set to POLICE. (see [openconfig type-queue-behavior](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-qos.html#type-queue-behavior))

`/qos/scheduler-policies/scheduler-policy/schedulers/scheduler/one-rate-two-color/violate-action/config/drop` is set to TRUE (ie: violate = packet drop)

That scheduler is then applied to an interface.  Packets that violate the policer are counted via `openconfig/system/controlplane/control-plane-traffic/ingress/qos/scheduler-policy/scheduler-statistics/scheduler[sequence]/state/violating-pkts`
Since the violate action is drop, this counter will count dropped packets, yes?

Reading the [OC qos docs](https://www.openconfig.net/docs/models/qos/), it seems a queue is meant to be created implicitly by the device, and this same queue is implicitly associated with the interface where the scheduler is applied.    When the scheduler (policier) drops packets, should the queue drop packet counter also increment?  Or should the queue counter only count TAIL,RED,WRED dropped packets?  It seems most useful for the queue counter to only count  TAIL,RED,WRED dropped packets?